Achievements/SelfMade.lua

```diff
@@ -19,34 +19,36 @@ end
 function self_made_achievement:Unregister()
 	self_made_achievement:UnregisterEvent("PLAYER_EQUIPMENT_CHANGED")
 end
+
+-- This function loops through a given tooltip when the "PLAYER_EQUIPMENT_CHANGED" event fires.
+-- If it finds the player's name in the tooltip it sets the variable to true and you do not fail the achievement, if false then the rest of the event fires, failing the player.
 local function isSelfCreated(...)
-	for i = 1, select("#", ...) do
-		local region = select(i, ...)
-		if region and region:GetObjectType() == "FontString" then
-			local text = region:GetText() -- string or nil
-			if text ~= nil then
-				local player_name, _ = UnitName("player")
-				if string.find(text, player_name) ~= nil then
-					return true
-				end
-			end
+	local player_found = false
+	for i = 1, GameTooltip:NumLines() do
+		local player = UnitName("player")
+		local mytext = _G["GameTooltipTextLeft" .. i]
+		local text = mytext:GetText()
+		if string.match(text, player) then
+			player_found = true
+			break
+		else
+			player_found = false
 		end
 	end
-	return false
+	return player_found
 end
 
 -- Register Definitions
+-- This executes on player equipping and item, then calls isSelfCreated() to check the tooltip for the player's name.
 self_made_achievement:SetScript("OnEvent", function(self, event, ...)
 	local arg = { ... }
 	if event == "PLAYER_EQUIPMENT_CHANGED" then
-		if arg[2] == true then
```

Comment:
Why remove this?

Reply:
For context, the second argument is true if the inventory slot is left unequiped. At a glance it seems like this could misfail if the player were to remove self-created equipment leaving that slot open

Reply:
it didn't break when i did, i couldn't decipher what it was used for. Its checking the 2nd arg of a array that is set locally, but not sure what that argument is suppose to be.

Reply:
upon testing that section, arg[2] always returns false, even when the achievement fails or passes. So im still not sure what its purpose is. I did forget to remove the local arg = { ... } when i removed that line.

Reply:
https://wowpedia.fandom.com/wiki/PLAYER_EQUIPMENT_CHANGED for the second arg

Reply:
'''
GameTooltip:SetInventoryItem("player", arg[1]) is there or what it is doing, im able to comment it out and get the result.
'''
In lua you can ignore all but the first function args or return values by ignoring indices.

Reply:
> https://wowpedia.fandom.com/wiki/PLAYER_EQUIPMENT_CHANGED for the second arg

right but im not sure that does to check if the item being place in the slot was created by the player:

"boolean - True when a slot becomes empty, false when filled."

So it will be true when the player takes the item off, but it wont even get into the next if statement either way unless the isSelfCreated() == false and that will only happen when they put on an item that isnt made by them.

In lua you can ignore all but the first function args or return values by ignoring indices.:
in this case the first item in the args is the useful one because the Item Name is always the first arg.

Thanks for the links and docs :D Helpful in understanding

Reply:
> Why remove this?

So, .... we need that section so when people decide to do SelfMade and then if they simply remove an item but never replace it throws and error. Thats why its there. Ill add it back in